### ISSUE TITLE
Dump file has changed to having too many short comment lines.

### ISSUE THREAD

**Issue by elementx54** [idaholab/moose#9747]
### Description of the enhancement or error report
_Reproduce issue:_ `> app-opt --dump AutoPositionsMultiApp`
The dump seems to be squishing the comments to the right side because `execute_on` is listing all of the possible options, instead of the default one. This introduces a bunch of lines with lots of whitespace at the beginning with a short comments. I've see a few where it was "one word per line" comments, like `[./<AutoPositionsMultiApp>]`. I think it is only on blocks that have the `execute_on` option. But I'm sure it could happen if the value of an option or the name of the option is _very_ long.

I think a comparison of what the --dump looks like before and what it looks like currenlty would help show what I mean: (You'll probably have to scroll horizontally to see the comments.)
*BEFORE*
```
      [./<AreaPostprocessor>]
        boundary           = (required)        # The list of boundary IDs from the mesh where this boundary
                                               # condition applies
        control_tags       =                   # Adds user-defined labels for accessing object parameters
                                               # via control logic.
                                               # Group: Advanced
        enable             = 1                 # Set the enabled status of the MooseObject.
                                               # Group: Advanced
        execute_on         = TIMESTEP_END      # Set to (nonlinear|linear|timestep_end|timestep_begin|custom)
                                               # to execute only at that moment 
        outputs            =                   # Vector of output names were you would like to restrict
                                               # the output of variables(s) associated with this object
                                               # Group: Advanced
        type               = AreaPostprocessor
        use_displaced_mesh = 0                 # Whether or not this object should use the displaced mesh
                                               # for computation.  Note that in the case this is true but
                                               # no displacements are provided in the Mesh block the undisplaced
                                               # mesh will still be used.
                                               # Group: Advanced
      [../]
```
*CURRENTLY*
```
      [./<AreaPostprocessor>]
        allow_duplicate_execution_on_initial = 0                                                                  # In the case where this
                                                                                                                  # UserObject is depended
                                                                                                                  # upon by an initial condition,
                                                                                                                  # allow it to be executed
                                                                                                                  # twice during the initial
                                                                                                                  # setup (once before the
                                                                                                                  # IC and again after mesh
                                                                                                                  # adaptivity (if applicable).
                                                                                                                  # Group: Advanced
        boundary                             = (required)                                                         # The list of boundary
                                                                                                                  # IDs from the mesh where
                                                                                                                  # this boundary condition
                                                                                                                  # applies
        control_tags                         =                                                                    # Adds user-defined labels
                                                                                                                  # for accessing object
                                                                                                                  # parameters via control
                                                                                                                  # logic.
                                                                                                                  # Group: Advanced
        enable                               = 1                                                                  # Set the enabled status
                                                                                                                  # of the MooseObject.
                                                                                                                  # Group: Advanced
        execute_on                           = 'none initial linear nonlinear timestep_end timestep_begin custom' # Set to (nonlinear|linear|timestep_end|timestep_begin|custom
                                                                                                                  # to execute only at that
                                                                                                                  # moment
        outputs                              =                                                                    # Vector of output names
                                                                                                                  # were you would like to
                                                                                                                  # restrict the output of
                                                                                                                  # variables(s) associated
                                                                                                                  # with this object
                                                                                                                  # Group: Advanced
        type                                 = AreaPostprocessor
        use_displaced_mesh                   = 0                                                                  # Whether or not this object
                                                                                                                  # should use the displaced
                                                                                                                  # mesh for computation.
                                                                                                                  # Note that in the case
                                                                                                                  # this is true but no displacements
                                                                                                                  # are provided in the Mesh
                                                                                                                  # block the undisplaced
                                                                                                                  # mesh will still be used.
                                                                                                                  # Group: Advanced
      [../]
```
### Rationale for the enhancement or information for reproducing the error
I'm pretty sure this is not the intended behavior and it doesn't look good. Also, if a dump was piped to a file, `app-opt --dump > file.dump` then the file will have more lines than previously. And I noticed this because I parse dump files for Vim for folding blocks and new lines are more expensive than just having longer lines.

### Identified impact
_(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)_
I'm not sure this would impact anything. However, the `execute_on` option's value seems to list all possible options, where before it was listing the default value.

**Comment by brianmoose** [comment]
The `execute_on` problem should be fixed and it does improve the looks of at least those blocks that used it.
If anybody has better ideas  on how to flow the parameter description comment with long name=value lines, please chime in.

**Comment by permcody** [comment]
Another option might be to print the description on a separate line altogether. Maybe indented, this format is for human consumption so whatever looks the best is fine with me.

**Comment by elementx54** [comment]
I personally like having it in a format that mirrors the input file so you can copy and paste directly and then start editing/deleting the things you want. The placement of the comments isn't too important; just that they remain comments.  

However, having them all lined up on the end allows for easy deletion after pasting using a rectangular selection (instead of character/line selection) or copying and pasting without the comments by using a rectangular selection.

**Comment by permcody** [comment]
closed by #9750
